pkg/registry/reference.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"strings"
+
+	orasregistry "oras.land/oras-go/pkg/registry"
+)
+
+type Reference struct {
+	OrasReference orasregistry.Reference
```

Comment:
Apologies in advance for the slow drip of comments

Do you think it is worthwhile not exposing this Oras V1 type directly? Especially given the experience in https://github.com/helm/helm/pull/13382 ?

Reply:
(the suggestion is to make `NewReference` and internal type (not exposed to the public Helm API). And so it won't be subject to API compatibility guarantees once merged.

Reply:
I made all that private to registry